tests/test_configuration.py

```diff
@@ -176,6 +176,27 @@ def test_get_item(config):
     assert config["ec_launchers"]["beams"]["steering_angles"]["waveform/1"] == waveform1
 
 
+def test_rename_waveform(config):
+    waveform1 = Waveform(name="waveform/1")
+    config.add_waveform(waveform1, ["ec_launchers"])
+    assert config["ec_launchers"]["waveform/1"] == waveform1
+
+    config.rename_waveform("waveform/1", "waveform/2")
+    with pytest.raises(KeyError):
+        assert config["ec_launchers"]["waveform/1"] == waveform1
```

Comment:
Test what you want to test :) 
```suggestion
    assert "waveform/1" not in config["ec_launchers"]
```